src/server.js

```diff
@@ -2,43 +2,61 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
  */
 
-// @flow
 /* eslint-env node */
 
+import bodyparser from 'koa-bodyparser';
+
 import {createPlugin, memoize} from 'fusion-core';
-import type {Context, FusionPlugin} from 'fusion-core';
-//$FlowFixMe
+import type {Context} from 'fusion-core';
 import {UniversalEventsToken} from 'fusion-plugin-universal-events';
-import bodyparser from 'koa-bodyparser';
+
 import MissingHandlerError from './missing-handler-error';
-import {BodyParserOptionsToken, RPCHandlersToken} from './tokens';
+import {BodyParserOptionsToken, RPCHandlersToken} from './tokens.js';
+import type {HandlerType} from './tokens.js';
+import type {RPCPluginType, IEmitter} from './types.js';
 
 const statKey = 'rpc:method';
 
 /* Helper function */
-function hasHandler(handlers, method) {
+function hasHandler(handlers: HandlerType, method: string): boolean {
   return handlers.hasOwnProperty(method);
 }
 
 class RPC {
-  ctx: Context;
-  emitter: *;
-  handlers: *;
+  ctx: ?Context;
+  emitter: ?IEmitter;
+  handlers: ?HandlerType;
+  fetch: ?*;
 
-  constructor(emitter: any, handlers: any, ctx: Context) {
+  constructor(emitter: any, handlers: any, ctx: Context): RPC {
     if (!ctx || !ctx.headers) {
       throw new Error('fusion-plugin-rpc requires `ctx`');
     }
     this.ctx = ctx;
     this.emitter = emitter;
     this.handlers = handlers;
+
+    return this;
   }
 
   async request(method: string, args: mixed) {
     const startTime = ms();
+
+    if (!this.ctx) {
```

Comment:
Why do we set Context to nullable on line 30 if we also have to add these error messages when doing so?

Reply:
`Context` is required for the server-side plugin, but not the client-side one.  If you look at interface for `RPCPluginType` it has properties that are only sometimes present as optional to account for this.  An alternative would be to have a server type and a client type, but that would violate our "register once and get it in both places" paradigm.

Reply:
Makes sense, thanks!